File: lib/java/src/main/java/io/github/msgpact/internal/types/VObject.java
package io.github.msgpact.internal.types;

import static io.github.msgpact.internal.generation.GenerateRandomObject.generateRandomObject;
import static io.github.msgpact.internal.validation.ValidateObject.validateObject;

import java.util.List;

import io.github.msgpact.internal.generation.GenerateContext;
import io.github.msgpact.internal.validation.ValidateContext;
import io.github.msgpact.internal.validation.ValidationFailure;

public class VObject implements VType {

    public static final String _OBJECT_NAME = "Object";

    @Override
    public int getTypeParameterCount() {
        return 1;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<VTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateObject(value, typeParameters, ctx);
    }

    @Override
    public Object generateRandomValue(Object blueprintValue, boolean useBlueprintValue,
            List<VTypeDeclaration> typeParameters, GenerateContext ctx) {
        return generateRandomObject(blueprintValue, useBlueprintValue, typeParameters, ctx);
    }

    @Override
    public String getName() {
        return _OBJECT_NAME;
    }

}

File: lib/ts/src/internal/types/VObject.ts
import { VTypeDeclaration } from './VTypeDeclaration';
import { ValidationFailure } from '../validation/ValidationFailure';
import { VType } from './VType';
import { validateObject } from '../validation/ValidateObject';
import { generateRandomObject } from '../generation/GenerateRandomObject';
import { GenerateContext } from '../generation/GenerateContext';
import { ValidateContext } from '../validation/ValidateContext';

export const objectName: string = 'Object';

export class VObject implements VType {
    getTypeParameterCount(): number {
        return 1;
    }

    validate(value: any, typeParameters: VTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateObject(value, typeParameters, ctx);
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: VTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomObject(blueprintValue, useBlueprintValue, typeParameters, ctx);
    }

    getName(): string {
        return objectName;
    }
}
